internal/daemon/cache/search.go

```diff
@@ -0,0 +1,155 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/api/sessions"
+	"github.com/hashicorp/boundary/api/targets"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/util"
+	"github.com/hashicorp/go-bexpr"
+)
+
+// SearchParams contains the paramaters for searching in the cache
+type SearchParams struct {
+	// the name of the resource. eg. "targets" or "sessions"
+	Resource string
```

Comment:
if you defined a `type SearchableResource string` and defined the valid enums
for it, you coiuld have type safety here and in the `searchableResources` map